src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataService.cs

```diff
@@ -138,116 +134,64 @@ public void InitializeService(ServiceHost serviceHost)
+        internal static async Task GetServerContextualization(GetServerContextualizationParams contextualizationParams, RequestContext<GetServerContextualizationResult> requestContext)
         {
             MetadataService.ConnectionServiceInstance.TryFindConnection(contextualizationParams.OwnerUri, out ConnectionInfo connectionInfo);
-
-            if (connectionInfo != null)
+            if (connectionInfo == null)
+            {
+                Logger.Error("Failed to find connection info about the server.");
+                await requestContext.SendError("Failed to find connection info about the server.");
+            }
+            else
             {
-                using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connectionInfo, "metadata"))
+                if (connectionInfo != null)
                 {
-                    // If scripts have been generated within the last 30 days then there isn't a need to go through the process
-                    // of generating scripts again.
                     if (MetadataScriptTempFileStream.IsScriptTempFileUpdateNeeded(connectionInfo.ConnectionDetails.ServerName))
                     {
-                        var scripts = SmoScripterHelpers.GenerateAllServerTableScripts(sqlConn)?.ToArray();
-                        if (scripts != null)
+                        using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connectionInfo, "metadata"))
                         {
-                            try
+                            var scripts = SmoScripterHelpers.GenerateAllServerTableScripts(sqlConn)?.ToArray();
+                            if (scripts != null)
                             {
-                                await requestContext.SendResult(new GenerateServerContextualizationResult()
+                                try
                                 {
-                                    Context = string.Join('\n', scripts)
-                                });
-
-                                MetadataScriptTempFileStream.Write(connectionInfo.ConnectionDetails.ServerName, scripts);
+                                    await requestContext.SendResult(new GetServerContextualizationResult()
+                                    {
+                                        Context = string.Join('\n', scripts)
+                                    });
+
+                                    MetadataScriptTempFileStream.Write(connectionInfo.ConnectionDetails.ServerName, scripts);
+                                }
+                                catch (Exception ex)
+                                {
+                                    Logger.Error($"An error was encountered while writing server contextualization scripts to the cache. Error: {ex.Message}");
```

Comment:
This isn't necessarily true, maybe just make this generic "while generating server contextualization scripts"